redaxo/src/addons/install/lib/package/package_update.php

```diff
@@ -4,17 +4,16 @@
  * @package redaxo\install
  *
  * @internal
+ *
+ * @psalm-suppress MissingConstructor
  */
 class rex_install_package_update extends rex_install_package_download
 {
     /**
-     * @var rex_addon|null
+     * @var rex_addon
```

Comment:
ist ein property was keinen default wert hat, oder im `__construct` initialisiert wird, immer auch `null`?

Reply:
Eigentlich ist die Property nullable.
Tatsächlich wird sie aber in der ersten Methode, die innerhalb dieser Subklasse aufgerufen wird, immer gesetzt.
Und ist somit in allen anderen Methoden dann auch immer verfügbar.

Deswegen ist hier mein pragmatischer Vorschlag, die als nicht-nullable zu setzen und oben MissingConstructor unterdrücken.

Nach gleichem Schema bin ich auch mit den Properties in rex_install_package_download vorgegangen.
Auch da werden sie in der ersten aufgerufenen Methode immer gesetzt.